dali/operators/image/convolution/gaussian_blur.cc

```diff
@@ -0,0 +1,305 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "dali/core/static_switch.h"
+#include "dali/kernels/imgproc/convolution/separable_convolution_cpu.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/operators/image/convolution/gaussian_blur.h"
+#include "dali/operators/image/convolution/gaussian_blur_params.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operator/common.h"
+
+namespace dali {
+
+using namespace gaussian_blur;  // NOLINT
+
+constexpr static const char* kSigmaArgName = "sigma";
+constexpr static const char* kWindowSizeArgName = "window_size";
+
+DALI_SCHEMA(GaussianBlur)
+    .DocStr(R"code(Apply Gaussian Blur to the input.
+
+User can specify sigma, kernel window size or both.
+If only the sigma is provided, the radius of kernel is calculated as ``ceil(3 * sigma)``,
+thus the kernel window size is ``2 * ceil(3 * sigma) + 1``.
+
+If only the kernel window size is provided, the sigma is calculated using the following formula::
+
+  radius = (window_size - 1) / 2
+  sigma = (radius - 1) * 0.3 + 0.8
+
+Both sigma and kernel window size can be specified as single value for all data axes
+or per data axis.
+
+When specifying the sigma or window size per axis, they are provided same as layouts: from outermost
+to innermost.
+The channel ``C`` and frame ``F`` dimensions are not considered data axes.
+If channels are present only channel-first or channel-last inputs are supported.
+
+For example, with ``HWC`` input, user can provide ``sigma=1.0`` or ``sigma=(1.0, 2.0)`` as there
+are two data axes H and W.
+
+The same input can be provided as per-sample tensors.
+)code")
+    .NumInput(1)
+    .NumOutput(1)
+    .AllowSequences()
+    .SupportVolumetric()
+    .AddOptionalArg<int>(kWindowSizeArgName, "The diameter of kernel.", std::vector<int>{0}, true)
+    .AddOptionalArg<float>(kSigmaArgName, "Sigma value for Gaussian Kernel.",
+                           std::vector<float>{0.f}, true)
+    .AddOptionalArg(
+        "dtype", "Output data type; if not set, the input type is used. Supported type: `FLOAT`.",
+        DALI_NO_TYPE);
+
+/**
+ * @brief Fill the result span with the argument which can be provided as:
+ * * ArgumentInput - {result.size()}-shaped Tensor
+ * * ArgumentInput - {1}-shaped Tensor, the value will be replicated `result.size()` times
+ * * Vector input - single "repeated argument" of length {result.size()} or {1}
+ * * scalar argument - it will be replicated `result.size()` times
+ *
+ * TODO(klecki): we may want to make this a generic utility and propagate the span-approach to
+ * the rest of the related argument gettters
+ */
+template <typename T>
+void GetGeneralizedArg(span<T> result, const std::string name, int sample_idx, const OpSpec& spec,
```

Comment:
Why not `Operator::GetPerSampleArgument`?

Reply:
It does something else - I need to obtain vectors per sample not scalars per sample.